concrete-core/src/specification/engines/glwe_key_tensor_product.rs

```diff
@@ -0,0 +1,48 @@
+use super::engine_error;
+use crate::specification::engines::AbstractEngine;
+use crate::specification::entities::GlweSecretKeyEntity;
+
+engine_error! {
+    GlweKeyTensorProductError for GlweKeyTensorProductEngine @
+    PolynomialSizeMismatch => "The polynomial size of the two input keys is not the same",
+    InputGlweDimensionMismatch => "The Glwe dimension of the two input keys is not the same"
+}
+
+impl<EngineError: std::error::Error> GlweKeyTensorProductError<EngineError> {
+    pub fn perform_generic_checks<InputKey1, InputKey2>(
+        input1: &InputKey1,
+        input2: &InputKey2,
+    ) -> Result<(), Self>
+    where
```

Comment:
GLWE key types:

1. Distributions of inputs1,2 do not need to match
2. Distribution of the output key could be some generic "Undefined" type